src/grpc/stream_infer_handler.cc

```diff
@@ -140,6 +140,17 @@ ModelStreamInferHandler::Process(InferHandler::State* state, bool rpc_ok)
   // This means that we only need to take care of
   // synchronizing this thread and the ResponseComplete
   // threads.
+  // We need an explicit finish indicator. Can't use 'state->step_'
+  // because we launch an async thread that could update 'state's
+  // step_ to be FINISH before this thread exits this function.
+  bool finished = false;
+  if (state->context_->grpc_strict_) {
+    std::lock_guard<std::recursive_mutex> lock(state->context_->mu_);
+    // Check if stream error detected and already connection ended
+    if (state->context_->IsGRPCStrictError()) {
+      return finished;
```

Comment:
For better or worse, I believe we generally return `!finished` (inverse) everywhere throughout the code, or a literal `false/true`. 

Is this line returning `finished` correct?